frontend/src/app/homepage-offer/offers.service.ts

```diff
@@ -1,37 +1,31 @@
-import { Offer } from './offers.model';
+import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
-import { Http, Response } from '@angular/http';
-import { Observable } from 'rxjs/Observable';
+
 import 'rxjs/add/operator/map';
+import { Observable } from 'rxjs/Observable';
 
 import { environment } from '../../environments/environment';
-
+import { Offer } from './offers.model';
 
 
 @Injectable()
 export class OffersService {
   private url = `${environment.apiRoot}/offers/`;
-  constructor (private http: Http) { }
-
-  getOffers() {
-    return this.http.get(this.url, { withCredentials: true } )
-      .map((res: Response) => res.json())
-      .map(offers => {
-        for (const offer of offers) {
-          this.loadDefaultImage(offer);
-        }
-        return offers;
-      });
+
+  constructor (private http: HttpClient) { }
+
+  getOffers(): Observable<Offer[]> {
+    return this.http.get<Offer[]>(this.url)
+      .map(offers => offers.map(offer => this.loadDefaultImage(offer)));
   }
 
   getOffer(id: number): Observable<Offer> {
-    return this.http.get(`${this.url}${id}/`, { withCredentials: true })
-      .map((res: Response) => res.json())
+    return this.http.get<Offer>(`${this.url}${id}/`)
       .map(offer => this.loadDefaultImage(offer));
   }
 
   loadDefaultImage(offer: Offer): Offer {
-    if (offer.image === null) {
+    if (!offer.image) {
```

Comment:
That's some kind of good habit?

Reply:
As this is also the style that I use extensively, I'll try to explain it. It might be considered as a good habit - cleaner and faster to read.  That's also the kind of codestyle "mistake" that would be pointed out to me in my job, during review process :)

No matter if offer.image is null or undefined, !offer.image will return true. 

Using explicit comparison e.g offer.image === null or offer.image === undefined is usefull only in a situation when we want to check if the value is null (or undefined), not whether it has been assigned any content, e.g:

if (offer.image === null) {
   alert('This offer does not contain the image');
} else if (offer.image === undefined) {
  alert('You never ever defined the image on this offer');
} else {
 alert('Your offer does have an image);
}

I have so far never written a code that would need specific differentiation between 'undefined' and 'null', but this might be a subject to change ;)

Reply:
Another way to protect it is to test all three possibilities :)
Image cannot be an object nor a list so we're safe here.